### ISSUE TITLE
FormFieldState value not in sync with the onChanged value from TextFormField

### ISSUE THREAD

**Issue by miguelpruivo** [flutter/flutter#36543]
## Issue
If you
```
TextFormField (
  onChanged: (value) {
                 print(value);
                 print("key value: " + _formFieldKey.currentState.value);
               });
```
You'll find that as the user types in, the value from the `onChanged` will _always_ be one character behind or ahead (if deleting) of the `_formFieldKey.currentState.value`.

This have an impact when you're validating on the go as the user types in, and thus, displaying an error even if the user has already a valid input, for example, if you're validating an email, when the user has typed *foo@company.c* it is already a valid input, but since the key value is displaying *foo@company.*, it provides a wrong information if you `validate()`.

Edit: It looks like it was working fine in the `stable 1.5.4`. 

## Flutter Doctor
```
[✓] Flutter (Channel unknown, v1.8.0, on Mac OS X 10.14.5 18F132, locale pt-PT)
    • Flutter version 1.8.0 at /Users/miguelruivo/DevTools/flutter
    • Framework revision 2fefa8c731 (3 weeks ago), 2019-07-01 11:33:22 -0700
    • Engine revision 45b66b722e
    • Dart version 2.4.0
```

**Comment by TahaTesser** [comment]
Hi @miguelpruivo 
Can you please if this issue still exist for you on latest dev channel?
if yes,  can you please provide your `flutter doctor -v` and `flutter run --verbose`? Also, to better address the issue, would be helpful
if you could post a minimal code sample to reproduce the problem
Thank you

**Comment by VladyslavBondarenko** [comment]
Reproduced. Complete code sample

```
import 'package:flutter/material.dart';

void main() => runApp(MaterialApp(home: MyApp()));

class MyApp extends StatefulWidget {
  @override
  _MyAppState createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  final _formFieldKey = GlobalKey<FormFieldState>();

  @override
  Widget build(BuildContext context) {
    return SafeArea(
      child: Scaffold(
        body: TextFormField (
          key: _formFieldKey,
            onChanged: (value) {
              print(value);
              print("key value: " + _formFieldKey.currentState.value);
            }),
      ),
    );
  }
}
```

<details>
 <summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel dev, v1.18.0-8.0.pre, on Mac OS X 10.15.4 19E287, locale en-GB)
    • Flutter version 1.18.0-8.0.pre at /Users/nevercode/dev/flutter_dev
    • Framework revision e0c63cd35e (4 days ago), 2020-04-23 22:29:01 -0400
    • Engine revision d2ec21221e
    • Dart version 2.9.0 (build 2.9.0-3.0.dev b0d35855d8)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/nevercode/Library/Android/sdk
    • Platform android-29, build-tools 29.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4.1, Build version 11E503a
    • CocoaPods version 1.9.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.44.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.9.1

 
[✓] Connected device (5 available)            
    • Android SDK built for x86 • emulator-5554                            • android-x86    • Android 10 (API 29) (emulator)
    • Nevercode’s iPhone        • b668e524315069f3db3661ac11ff1f66afafebdb • ios            • iOS 13.4.5
    • macOS                     • macOS                                    • darwin-x64     • Mac OS X 10.15.4 19E287
    • Web Server                • web-server                               • web-javascript • Flutter Tools
    • Chrome                    • chrome                                   • web-javascript • Google Chrome 81.0.4044.122

• No issues found!
```
</details>

**Comment by tvolkert** [comment]
Fixed by #65695
